test/e2e/multiple_zk_test.go

```diff
@@ -0,0 +1,187 @@
+	defaultCluster.Spec.Persistence.VolumeReclaimPolicy = "Delete"
+	defaultCluster.ObjectMeta.Name = "zk2"
+	initialVersion := "0.2.5"
+	upgradeVersion := "0.2.7"
+	defaultCluster.Spec.Image = api.ContainerImage{
+		Repository: "pravega/zookeeper",
+		Tag:        initialVersion,
+	}
+	zk2, err := zk_e2eutil.CreateCluster(t, f, ctx, defaultCluster)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	err = zk_e2eutil.WaitForClusterToBecomeReady(t, f, ctx, zk2, podSize)
+	g.Expect(err).NotTo(HaveOccurred())
+	// This is to get the latest Pravega cluster object
+	zk2, err = zk_e2eutil.GetCluster(t, f, ctx, zk2)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	g.Expect(zk2.Status.CurrentVersion).To(Equal(initialVersion))
+
+	defaultCluster = zk_e2eutil.NewDefaultCluster(namespace)
+
+	defaultCluster.WithDefaults()
+	defaultCluster.Status.Init()
+	defaultCluster.ObjectMeta.Name = "zk3"
+	defaultCluster.Spec.Persistence.VolumeReclaimPolicy = "Delete"
+
+	zk3, err := zk_e2eutil.CreateCluster(t, f, ctx, defaultCluster)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	err = zk_e2eutil.WaitForClusterToBecomeReady(t, f, ctx, zk3, podSize)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// This is to get the latest zk cluster object
+	zk1, err = zk_e2eutil.GetCluster(t, f, ctx, zk1)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// scale up the replicas in first cluster
+	zk1.Spec.Replicas = 5
+	podSize = 5
+
+	err = zk_e2eutil.UpdateCluster(t, f, ctx, zk1)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	err = zk_e2eutil.WaitForClusterToBecomeReady(t, f, ctx, zk1, podSize)
+	g.Expect(err).NotTo(HaveOccurred())
+	//scale down the replicas back to 3
+
+	zk1, err = zk_e2eutil.GetCluster(t, f, ctx, zk1)
+	g.Expect(err).NotTo(HaveOccurred())
+	zk1.Spec.Replicas = 3
+	podSize = 3
+
+	err = zk_e2eutil.UpdateCluster(t, f, ctx, zk1)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	err = zk_e2eutil.WaitForClusterToBecomeReady(t, f, ctx, zk1, podSize)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// This is to get the latest zk cluster object
+	zk2, err = zk_e2eutil.GetCluster(t, f, ctx, zk2)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	//upgrade the image in second Cluster
+	zk2.Spec.Image.Tag = upgradeVersion
+
+	err = zk_e2eutil.UpdateCluster(t, f, ctx, zk2)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	err = zk_e2eutil.WaitForClusterToUpgrade(t, f, ctx, zk2, upgradeVersion)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// This is to get the latest Pravega cluster object
+	zk2, err = zk_e2eutil.GetCluster(t, f, ctx, zk2)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	g.Expect(zk2.Spec.Image.Tag).To(Equal(upgradeVersion))
+	g.Expect(zk2.Status.CurrentVersion).To(Equal(upgradeVersion))
+	g.Expect(zk2.Status.TargetVersion).To(Equal(""))
+
+	// This is to get the latest zk cluster object
+	zk3, err = zk_e2eutil.GetCluster(t, f, ctx, zk3)
+	g.Expect(err).NotTo(HaveOccurred())
+	//Delete all pods in the 3rd Cluster
+	podDeleteCount := 3
+	err = zk_e2eutil.DeletePods(t, f, ctx, zk3, podDeleteCount)
+	g.Expect(err).NotTo(HaveOccurred())
```

Comment:
Here you are adding sleep for 60 seconds to ensure pods ready has become false, in place of sleep can we set the pods Ready condition to false?

Reply:
I have tried setting pods Ready condition to false, but if we go GetCluster immediately after this, pods Ready ciondition is set to True, and Cluster ready returns immediately.

Reply:
Okay, so sleep would do it, for now, thanks for checking it.